dynawo/sources/Solvers/SolverKINSOL/DYNSolverKIN.cpp

```diff
@@ -147,23 +162,25 @@ SolverKIN::init(const shared_ptr<Model>& model, modeKin_t mode) {
 
   // (8) Solver choice
   // -------------------
-  flag = KINKLU(KINMem_, nbF_, 10., CSR_MAT);
+  M_ = SUNSparseMatrix(nbF_, nbF_, 10., CSR_MAT);
+  LS_ = SUNLinSol_KLU(yy_, M_);
```

Comment:
We could also deal with potential problems here when creating the structures as for IDA, ie testing that M_ != NULL and the same for LS and displaying some error messages at this stage if something went wrong

Reply:
Solved